procrastinate/testing.py

```diff
@@ -192,17 +192,24 @@ def finished_jobs(self) -> list[JobRow]:
     def fetch_job_one(self, queues: Iterable[str] | None) -> dict:
         # Creating a copy of the iterable so that we can modify it while we iterate
 
-        for job in self.jobs.values():
+        filtered_jobs = [
+            job
+            for job in self.jobs.values()
             if (
                 job["status"] == "todo"
                 and (queues is None or job["queue_name"] in queues)
                 and (not job["scheduled_at"] or job["scheduled_at"] <= utils.utcnow())
                 and job["lock"] not in self.current_locks
-            ):
-                job["status"] = "doing"
-                self.events[job["id"]].append({"type": "started", "at": utils.utcnow()})
+            )
+        ]
+
+        filtered_jobs.sort(key=lambda job: job["priority"], reverse=True)
 
-                return job
+        if filtered_jobs:
+            job = filtered_jobs[0]
+            job["status"] = "doing"
+            self.events[job["id"]].append({"type": "started", "at": utils.utcnow()})
+            return job
 
         return {"id": None}
```

Comment:
I think it's slightly more readable the other way around:
```python
if filtered_jobs:
    return {"id": None} 
```

This lets more code avoid indentation. As a general rule, and with possible exceptions, I tend to try and identify the special case that I can handle first (especially if they allow returning early) and try to have the general case be the one that is as little indented as possible, and goes all the way to the bottom of the function. This avoids the "tower of if-statements" when possible.

Reply:
Sounds good. I am always in favor when it comes to improving code structure.